### PR TITLE
feat: Client Tools

### PR DESCRIPTION
**Pull request by louisjoecodes**
Add support for client-tools in Python

### CODE REVIEW

File: src/elevenlabs/conversational_ai/conversation.py [hunk 1]
```diff
@@ -52,22 +52,44 @@ def interrupt(self):
         """
         pass
 
+
+class ClientTools:
+    def __init__(self):
+        self.tools = {}
+        self.lock = threading.Lock()
+
+    def register(self, tool_name, handler):
```
Comment (lacop11) [inline src/elevenlabs/conversational_ai/conversation.py:61]: Can you add type annotation here please?

Also given the call blocks the processing of audio maybe add a comment saying the tool call should finish quickly. I think eventually we should turn this interface async and support async tools, then we can return the response when ready.

File: tests/e2e_test_convai.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    # Get API key from environment
+    api_key = os.getenv("ELEVENLABS_API_KEY")
+    if not api_key:
+        raise ValueError("Please set ELEVENLABS_API_KEY environment variable")
+
+    # Initialize client
+    client = ElevenLabs(api_key=api_key)
+
+    AGENT_ID = "<insert-testing-agent-id>"
```
Comment (lacop11) [inline tests/e2e_test_convai.py:22]: Take from env var maybe?

**Review by lacop11 — APPROVED** [review]
Nice, thanks for adding the tests!

File: src/elevenlabs/conversational_ai/conversation.py (hunk 1 above)
Comment (louisjoecodes) [reply src/elevenlabs/conversational_ai/conversation.py]: Good shout! added.

**Pull request merged**